dbustestd/tst_dbus_interface.qml

```diff
@@ -0,0 +1,261 @@
+/*****************************************************************************
+**
+** Copyright (C) 2015 Jolla Ltd.
+** Contact: Simo Piiroinen <simo.piiroinen@jollamobile.com>
+** All rights reserved.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+******************************************************************************/
+import QtQuick 2.0
+import Sailfish.Silica 1.0
```

Comment:
You could consider using QML TestCase, see http://doc.qt.io/qt-5/qml-qttest-testcase.html#details

That said, making asynchronous test cases with TestCase may require some extra work.